core/src/main/scala/anorm/package.scala

```diff
@@ -22,6 +22,16 @@ package object anorm {
   /** Structural type for timestamp wrapper. */
   type TimestampWrapper1 = { def getTimestamp: java.sql.Timestamp }
 
+  object TimestampWrapper1 {
+    import scala.language.reflectiveCalls
+
+    def unapply(that: Any): Option[java.sql.Timestamp] = try {
+      Some(that.asInstanceOf[TimestampWrapper1].getTimestamp)
+    } catch {
+      case _: Throwable => None
```

Comment:
I would much prefer we only catch `NoSuchMethodException` here (which I think is the only thing we'd expect this to throw).